crates/nargo/src/cli/prove_cmd.rs

```diff
@@ -52,27 +58,16 @@ pub(crate) fn run(args: ProveCommand, config: NargoConfig) -> Result<(), CliErro
         None
     };
 
-    prove_with_path(
-        args.proof_name,
-        config.program_dir,
-        proof_dir,
-        circuit_build_path,
-        args.verify,
-        args.show_ssa,
-        args.allow_warnings,
-    )?;
+    prove_with_path(config.program_dir, proof_dir, circuit_build_path, args)?;
```

Comment:
I'm not a fan of passing the `args` field wholesale. We're planning on creating `nargo-core` and `nargo-cli` crates in future so it would be good to maintain some separation between a CLI arg struct and the interface for a potential "core" function.